packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -184,58 +184,31 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange>
 			rebaseLogger,
 		);
 		this.editManager.localBranch.events.on("transactionStarted", () => {
-			this.commitEnricher.startNewTransaction();
+			this.commitEnricher.startTransaction();
 		});
 		this.editManager.localBranch.events.on("transactionAborted", () => {
-			this.commitEnricher.abortCurrentTransaction();
+			this.commitEnricher.abortTransaction();
 		});
 		this.editManager.localBranch.events.on("transactionCommitted", () => {
-			this.commitEnricher.commitCurrentTransaction();
+			this.commitEnricher.commitTransaction();
 		});
 		this.editManager.localBranch.events.on("beforeChange", (change) => {
-			// Ensure that any previously prepared commits that have not been sent are purged.
-			this.commitEnricher.purgePreparedCommits();
-			if (this.detachedRevision !== undefined) {
-				// Edits submitted before the first attach do not need enrichment because they will not be applied by peers.
-			} else if (change.type === "append") {
-				if (this.getLocalBranch().isTransacting()) {
-					for (const newCommit of change.newCommits) {
-						this.commitEnricher.ingestTransactionCommit(newCommit);
-					}
-				} else {
-					for (const newCommit of change.newCommits) {
-						this.commitEnricher.prepareCommit(newCommit, false);
-					}
-				}
-			} else if (
-				change.type === "replace" &&
-				getChangeReplaceType(change) === "transactionCommit" &&
-				!this.getLocalBranch().isTransacting()
-			) {
-				assert(
-					change.newCommits.length === 1,
-					0x983 /* Unexpected number of commits when committing transaction */,
-				);
-				this.commitEnricher.prepareCommit(change.newCommits[0] ?? oob(), true);
-			}
+			this.commitEnricher.processChange(change, this.detachedRevision === undefined);
```

Comment:
Can we just avoid calling this when `this.detachedRevision === undefined` is true? I find the logic of calling it but having it do almost nothing kind of confusing.

Reply:
No, because even though it does _almost_ nothing in that case, it still does something important - it nulls out the `#getOuterTransactionChange` property. That behavior is the reason that we don't have to pass in `concludesOuterTransaction` anymore. I don't love it either but I couldn't think of anything simpler.

Reply:
As discussed offline, I was able to remove the parameter by ensuring that we never attach during a transaction. With that guarantee, we can forgo doing _any_ updates to the commit enricher before we have attached.